pkg/daemon/ceph/osd/volume.go

```diff
@@ -613,20 +613,25 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 			"json",
 		}...)
 
-		cvOut, err := context.Executor.ExecuteCommandWithCombinedOutput(baseCommand, reportArgs...)
-		if err != nil {
-			return errors.Wrapf(err, "failed ceph-volume json report: %s", cvOut) // fail return here as validation provided by ceph-volume
-		}
+		cephVersion := a.clusterInfo.CephVersion
 
-		logger.Debugf("ceph-volume report: %+v", cvOut)
+		if !cephVersion.IsNautilus() || cephver.IsInferior(cephVersion, cephver.CephVersion{Major: 14, Minor: 2, Extra: 13}) {
```

Comment:
Could you add a comment to this line explaining why it's needed for when we come back to it in 6-12 months?

Reply:
Instead of checking for the version and skipping it if a broken version, what if we always run the report, but just log and ignore any errors that it produces? Seems like a good pattern so we can always attempt to create the OSDs even if the report fails.

Reply:
I think it's possible, but maybe a new type should be introduced to reflect the new return structure...

Reply:
What new type are you referring to?

Reply:
Before 14.2.13, it looks like

```json
{"changed": true, "vg": {"devices": xxx}}
```

For 14.2.15, it looks like

```console
2020-12-14 02:15:28.771476 D | exec: Running command: stdbuf -oL ceph-volume --log-path /tmp/ceph-log lvm batch --prepare --bluestore --yes --osds-per-device 1 --crush-device-class sjt /dev/vdc --db-devices /dev/vdd --report --format json
2020-12-14 02:15:37.640274 D | cephosd: ceph-volume report: --> passed data devices: 1 physical, 0 LVM
--> relative data size: 1.0
--> passed block_db devices: 1 physical, 0 LVM
[{"block_db": "/dev/vdd", "encryption": "None", "data": "/dev/vdc", "data_size": "100.00 GB", "block_db_size": "100.00 GB"}]
```

Reply:
Octopus v15.2.8 was just released and the [release notes](https://ceph.io/releases/v15-2-8-octopus-released/) indicated it also received an update to ceph-volume batch mode, so I assume we also need the same change on v15.2.8 and newer.